Delay initial data loading (embedded JS in the HTML)

joakin opened the pull request:
Let's not duplicate the initial html size with the initial JS data and load it from a script tag instead.

joakin committed:
Load JS initial store data from script tag

To avoid having the initial html payload size duplicated.

joakin committed:
Inline data when small enough in initial HTML payload

To avoid a double roundtrip when the payload is not worth it (smaller than
30kb) (magic number?).

phuedx commented on lib/server/routes.js:
```diff
@@ -64,6 +53,40 @@ router.get('*', (req, res, next) => {
+        url: req.url,
+        // Send inline data when it is smaller than 30kb. Else let the browser
+        // request it with a script tag
+        data: state.length < 30 * 1024 ? state : undefined
```
It'd be nice to trigger this with a query parameter but this'll do for now. Also, 30 KB is going to be larger than most lead sections, so this – rather nice (!) – optimisation won't be invoked for most initial requests.

phuedx commented:
If you were to make this configurable now, then we could compare the two approaches easily with WPT in a little while.

joakin committed:
Allow overriding inline-data strategy with query param

So that we can test when it makes sense and when it doesn't to tweak the magic
number.

joakin commented:
@phuedx  See 33ca014 ^

phuedx commented:
![pew pew](http://media0.giphy.com/media/11xzhhh5mCmCju/giphy.gif)
